controllers/azurejson_machinepool_controller_test.go

```diff
@@ -210,3 +217,140 @@ func TestAzureJSONPoolReconciler(t *testing.T) {
 		})
 	}
 }
+
+func TestAzureJSONPoolReconcilerUserAssignedIdentities(t *testing.T) {
+	g := NewWithT(t)
+	ctrlr := gomock.NewController(t)
+	defer ctrlr.Finish()
+	req := ctrl.Request{NamespacedName: types.NamespacedName{Name: "fake-machine-pool", Namespace: "fake-ns"}}
+	ctx := context.Background()
+	scheme := runtime.NewScheme()
+
+	err := infrav1exp.AddToScheme(scheme)
+	g.Expect(err).ToNot(HaveOccurred())
+
+	err = expv1.AddToScheme(scheme)
+	g.Expect(err).ToNot(HaveOccurred())
+
+	err = corev1.AddToScheme(scheme)
+	g.Expect(err).ToNot(HaveOccurred())
+
+	err = clusterv1.AddToScheme(scheme)
+	g.Expect(err).ToNot(HaveOccurred())
+
+	err = infrav1.AddToScheme(scheme)
+	g.Expect(err).ToNot(HaveOccurred())
+
+	azureMP := &infrav1exp.AzureMachinePool{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "fake-machine-pool",
+			Namespace: "fake-ns",
+			Labels: map[string]string{
+				clusterv1.ClusterNameLabel: "fake-cluster",
+			},
+			OwnerReferences: []metav1.OwnerReference{
+				{
+					APIVersion: expv1.GroupVersion.Group + "/" + expv1.GroupVersion.Version,
```

Comment:
nit: I think the convention in tests is to use fmt.Sprintf for concats. Example: https://github.com/willie-yao/cluster-api-provider-azure/blob/aks-fleet/api/v1beta1/azuremachine_validation_test.go#L201

Reply:
Utilized `fmt.Sprintf()` convention as suggested.  Resolving.